test/e2e/multihoming.go

```diff
@@ -627,254 +628,312 @@ var _ = Describe("Multi Homing", func() {
 			),
 		)
 
-		table.DescribeTable(
-			"multi-network policies configure traffic allow lists",
-			func(netConfig networkAttachmentConfig, allowedClientPodConfig podConfiguration, blockedClientPodConfig podConfiguration, serverPodConfig podConfiguration, policy *mnpapi.MultiNetworkPolicy) {
-				netConfig.namespace = f.Namespace.Name
-				blockedClientPodConfig.namespace = f.Namespace.Name
-				allowedClientPodConfig.namespace = f.Namespace.Name
-				serverPodConfig.namespace = f.Namespace.Name
+		Context("multi-network policies", func() {
+			const (
+				generatedNamespaceNamePrefix = "pepe"
+			)
+			var extraNamespace *v1.Namespace
 
-				By("creating the attachment configuration")
-				_, err := nadClient.NetworkAttachmentDefinitions(f.Namespace.Name).Create(
+			BeforeEach(func() {
+				createdNamespace, err := cs.CoreV1().Namespaces().Create(
 					context.Background(),
-					generateNAD(netConfig),
+					&v1.Namespace{
+						ObjectMeta: metav1.ObjectMeta{
+							Labels:       map[string]string{"role": "trusted"},
+							GenerateName: generatedNamespaceNamePrefix,
+						},
+					},
 					metav1.CreateOptions{},
 				)
 				Expect(err).NotTo(HaveOccurred())
+				extraNamespace = createdNamespace
+			})
 
-				By("instantiating the server pod")
-				serverPod, err := cs.CoreV1().Pods(serverPodConfig.namespace).Create(
+			AfterEach(func() {
+				blockUntilEverythingIsGone := metav1.DeletePropagationForeground
+				Expect(cs.CoreV1().Namespaces().Delete(
 					context.Background(),
-					generatePodSpec(serverPodConfig),
-					metav1.CreateOptions{},
-				)
-				Expect(err).NotTo(HaveOccurred())
-				Expect(serverPod).NotTo(BeNil())
+					extraNamespace.Name,
+					metav1.DeleteOptions{PropagationPolicy: &blockUntilEverythingIsGone},
+				)).To(Succeed())
+				Eventually(func() bool {
+					_, err := cs.CoreV1().Namespaces().Get(context.Background(), extraNamespace.Name, metav1.GetOptions{})
+					nsPods, podCatchErr := cs.CoreV1().Pods(extraNamespace.Name).List(context.Background(), metav1.ListOptions{})
+					return podCatchErr == nil && errors.IsNotFound(err) && len(nsPods.Items) == 0
+				}, 2*time.Minute, 5*time.Second).Should(BeTrue())
+			})
 
-				By("asserting the server pod reaches the `Ready` state")
-				Eventually(func() v1.PodPhase {
-					updatedPod, err := cs.CoreV1().Pods(serverPodConfig.namespace).Get(context.Background(), serverPod.GetName(), metav1.GetOptions{})
-					if err != nil {
-						return v1.PodFailed
+			table.DescribeTable(
+				"multi-network policies configure traffic allow lists",
+				func(netConfig networkAttachmentConfig, allowedClientPodConfig podConfiguration, blockedClientPodConfig podConfiguration, serverPodConfig podConfiguration, policy *mnpapi.MultiNetworkPolicy) {
+					if blockedClientPodConfig.namespace == "" {
+						blockedClientPodConfig.namespace = f.Namespace.Name
+					} else {
+						blockedClientPodConfig.namespace = extraNamespace.Name
```

Comment:
the thing is I want to use a different "extra" namespace on every iteration. Since this is a table, I do not have the name of the namespace available from the table entries.

Maybe I could transform this attribute from `namespace` to a boolean named `generateUniqueNamespaceName` ? At least that would be more explicit. 

Thoughts @jcaamano ?

Reply:
A boolean would make more sense indeed

Reply:
Like `useExtraNamespace` or such

Reply:
I can either do that on a follow-up PR (if we want to minimize the time the feature is merged upstream w/out e2e tests), or, if we wait a couple of days (until next Wednesday), I can adapt this PR with this.

I agree it makes more sense, and would be clearer.